src/shared/metrics.c

```diff
@@ -221,7 +194,7 @@ int metric_build_send(MetricFamilyContext* context, const char *object, sd_json_
         if (r < 0)
                 return r;
 
-        if (fields) { /* NULL => no fields object, empty strv => field:{} */
```

Comment:
Same here, you're fixing stuff from the first commit in the second commit, it should be fixed in the first commit.